### PR TITLE
enhance join arguments generation logic using template

### PR DESCRIPTION
**Pull request by bruceauyeung**
**What this PR does / why we need it**:
this PR enhances kubeadm join arguments generation logic using template, this makes code more readable and adding arguments more  easier.

**Which issue this PR fixes** _(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR gets merged)_: fixes #

**Special notes for your reviewer**:

**Release note**:

``` release-note
```

Signed-off-by: 欧阳钦华10079130 ouyang.qinhua@zte.com.cn

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35265)

### CODE REVIEW

**Comment by redhatlinux10** [comment]
I signed it!

**Comment by redhatlinux10** [comment]
hi, i have already signed CNCF, why this PR still say that redhatlinux10 needs CLA ?
please help!

**Review by jbeda — CHANGES_REQUESTED** [review]

**Comment by redhatlinux10** [comment]
hi @jbeda  , i have modified the codes as you suggested, would you like to review again ?

File: cmd/kubeadm/app/cmd/init.go [hunk 1]
```diff
@@ -36,6 +37,18 @@ import (
 	netutil "k8s.io/kubernetes/pkg/util/net"
 )
 
+const (
+	joinArgsTemplateLiteral = `--token={{.Cfg.Secrets.GivenToken -}}
+		{{if ne .Cfg.API.BindPort .DefaultAPIBindPort -}}
+		{{" --api-port="}}{{.Cfg.API.BindPort -}}
+		{{end -}}
+		{{if ne .Cfg.Discovery.BindPort .DefaultDiscoveryBindPort -}}
+		{{" --discovery-port=="}}{{.Cfg.Discovery.BindPort -}}
```
Comment (jbeda) [inline cmd/kubeadm/app/cmd/init.go]: I think there is an extra `=` here.

**Review by jbeda — CHANGES_REQUESTED** [review]
Looking good!  Just one typo.

**Comment by redhatlinux10** [comment]
@jbeda sorry for the typo, i have fixed it, please review, thanks !

**Review by jbeda — APPROVED** [review]

**Comment by jbeda** [comment]
Looks good!  Thanks!

**Pull request merged**